Supply chain: new SSC and security features (#16679)
docker/docs: scout/artifactory.md

---
description: >
  Integrate JFrog Artifactory and JFrog Container Registry with Docker Scout
keywords: >
  docker scout, jfrog, artifactory, jcr, integration, image analysis, security,
  cves
title: Artifactory integration
---

> **Note**
>
> Docker Scout is an [early access](../release-lifecycle.md#early-access-ea)

> > product, and requires a Pro, Team, or Business subscription.

Reviewer: Here I think it might be simpler to say "...early access product, and requires a subscription" since user will need a core Docker subscription, plus a subscription to Scout according to their level of use,

>

> > If you're interested this integration for your organization and want to learn

Reviewer 2: ```suggestion
> If you're interested in this integration for your organization and want to learn
```

> learn more, get in touch by filling out the contact form on the
> [Docker Scout product page](https://docker.com/products/scout){:
> target="\_blank" rel="noopener" }.

Integrating Docker Scout with JFrog Artifactory lets you run image analysis
automatically on images in your Artifactory registries.

This integration is made possible by a monitoring agent. The agent is a
standalone service that analyzes images and uploads the result to Docker Scout.
You can view the results using the
[Docker Scout web UI](https://dso.docker.com/){: target="\_blank" rel="noopener"
}.

## How it works

The Docker Scout Artifactory agent is available as an
[image on Docker Hub](https://hub.docker.com/r/docker/artifactory-agent){:
target="\_blank" rel="noopener" }. The agent works by continuously polling
Artifactory for new images. When it finds a new image, it performs the following
steps:

1. Pull the image from Artifactory
2. Analyze the image
3. Upload the analysis result to Docker Scout

The agent records the Software Bill of Material (SBOM) for the image, and the

> SBOMs for all of it's base images. The recorded SBOMs include both OS-level and

Reviewer 3: ```suggestion
SBOMs for all of its base images. The recorded SBOMs include both Operating System (OS)-level and
```

application-level programs or dependencies that the image contains.

Additionally, metadata about the image itself is also recorded:

- The source repository for the image
- Build instructions
- Build date
- Tags and digest
- Target platforms
- Layer sizes

> The agent transacts this data to Docker Scout. The agent never transacts the

Reviewer 3: "transacts"? Send maybe?

image itself, nor any data inside the image, such as code, binaries, and layer
blobs.

The agent doesn't detect and analyze pre-existing images. It only analyzes
images that appear in the registry while the agent is running.

## Deploy the agent

This section describes the steps for deploying the Artifactory agent.

### Prerequisites

> Before you deploy the agent, ensure that the prerequisites are met:

Reviewer 3: ```suggestion
Before you deploy the agent, ensure that you meet the prerequisites:
```

I see active voice in other places, so I _assume_ it's preferred…

| --------------------------- | ------------------------------------------------------------------------------- |
| `agent_id`                  | Unique identifier for the agent.                                                |
| `docker.organization_name`  | Name of the Docker organization.                                                |
| `docker.username`           | Username of the admin user in the Docker organization.                          |
| `docker.pat`                | Personal access token of the admin user with read and write permissions.        |
| `artifactory.base_url`      | Base URL of the Artifactory instance.                                           |
| `artifactory.username`      | Username of the Artifactory user with read permissions that the agent will use. |
| `artifactory.password`      | Password or API token for the Artifactory user.                                 |
| `artifactory.image_filters` | Optional: List of repositories and images to analyze.                           |

If you don't specify any repositories in `artifactory.image_filters`, the agent

> will run image analysis on all images in your Artifactory instance.

Reviewer 2: ```suggestion
runs image analysis on all images in your Artifactory instance.
```

The following snippet shows a sample configuration:

```json
{
  "agent_id": "acme-prod-agent",
  "docker": {
    "organization_name": "acme",
    "username": "mobythewhale",
    "pat": "dckr_pat__dsaCAs_xL3kNyupAa7dwO1alwg"
  },
  "artifactory": [
    {
      "base_url": "https://acme.jfrog.io",
      "username": "acmeagent",
      "password": "hayKMvFKkFp42RAwKz2K",
      "image_filters": [
        {
          "repository": "dev-local",
          "images": ["internal/repo1", "internal/repo2"]
        },
        {
          "repository": "prod-local",
          "images": ["staging/repo1", "prod/repo1"]
        }
      ]
    }
  ]
}
```

> Create a configuration file and save it on somewhere on the server where you

Reviewer 2: ```suggestion
Create a configuration file and save it somewhere on the server where you
```

plan to run the agent. For example, `/var/opt/artifactory-agent/config.json`.

### Run the agent

> The following example show how to run the Docker Scout Artifactory agent using

Reviewer 2: ```suggestion
The following example shows how to run the Docker Scout Artifactory agent using
```

`docker run`. This command creates a bind mount for the directory containing the
JSON configuration file created earlier to `/opt/artifactory-agent/data` inside
the container. Make sure the mount path you use is the directory containing the
`config.json` file.

<!-- prettier-ignore -->
> **Important**
>
> Use the `v1` tag of the Artifactory agent image. Don't use the `latest` tag as
> doing so may incur breaking changes.
{: .important }

```console
$ docker run \
  --mount type=bind,src=/var/opt/artifactory-agent,target=/opt/artifactory-agent/data \
  docker/artifactory-agent:v1
```

## View analysis results

You can view the image analysis results in the Docker Scout web UI.

1. Go to [Docker Scout web UI](https://dso.docker.com).
2. Sign in using your Docker ID.

   Once signed in, you're taken to the **Images** page. This page displays the

>    repositories in you organization connected to Docker Scout.

Reviewer 2: ```suggestion
   repositories in your organization connected to Docker Scout.
```
